dsl-interpreter/src/workflows.ts

```diff
@@ -0,0 +1,57 @@
+import { proxyActivities } from '@temporalio/workflow';
+// Only import the activity types
+import type * as activities from './activities';
+
+export type DSL = {
+  variables: Record<string, string>; // could be string | number?
+  root: Statement;
+};
+
+type Sequence = {
+  elements: Statement[];
+};
+type ActivityInvocation = {
+  name: string;
+  arguments?: string[];
+  result?: string;
+};
+type Parallel = {
+  branches: Statement[];
+};
+
+type Statement = { activity: ActivityInvocation } | { sequence: Sequence } | { parallel: Parallel };
+
+const acts = proxyActivities<typeof activities>({
+  startToCloseTimeout: '1 minute',
+}) as Record<string, (...args: string[]) => Promise<string | undefined>>;
+
+/** A workflow that simply calls an activity */
+export async function DSLInterpreter(dsl: DSL): Promise<any> {
+  const bindings = dsl.variables as Record<string, string>;
+  return execute(dsl.root, bindings);
```

Comment:
For a better stack trace:
```suggestion
  return await execute(dsl.root, bindings);
```

Reply:
ohh thats why you do it. i always wondered why you do that since the await is optional from an execution perspective.